### PR TITLE
fix peer test bug

### PR DESCRIPTION
**Pull request by nogang**
## Proposed changes

- `maybe` is a function that must return true with a 50% probability. However, the code before it was supposed to always return false.

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: networks/p2p/peer_test.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func TestPeerDisconnect(t *testing.T) {
 // This test is supposed to verify that Peer can reliably handle
 // multiple causes of disconnection occurring at the same time.
 func TestPeerDisconnectRace(t *testing.T) {
-	maybe := func() bool { return rand.Intn(1) == 1 }
+	maybe := func() bool { return rand.Intn(2) == 1 }
```
Comment (kjhman21) [inline networks/p2p/peer_test.go]: rand.intn(1) returns 0 or 1.
Please refer to https://golang.org/pkg/math/rand/#Intn

Comment (kjhman21) [reply networks/p2p/peer_test.go]: Sorry. It returns 0 always becaus it returns [0,1).
Please ignore this comment. :)

Comment (nogang) [reply networks/p2p/peer_test.go]: `rand.intn(1)` always returns 0. It open about `n`. 

`Mathematical interval notation such as [0, n) is used throughout the documentation for this package.` by https://golang.org/pkg/math/rand/#Intn

**Comment by KimKyungup** [comment]
I think if you link the ethereum PR, it is better. :)

**Pull request merged**